service/resource/deploymentv3/update.go

```diff
@@ -73,37 +73,50 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 		return nil, microerror.Mask(err)
 	}
 
-	var deploymentsToUpdate []*v1beta1.Deployment
-
 	if updateallowedcontext.IsUpdateAllowed(ctx) {
 		r.logger.LogCtx(ctx, "debug", "finding out which deployments have to be updated")
 
-		// Check if deployments changed. In case they did, add the deployments to
-		// the list of deployments intended to be updated, but only in case they are
-		// not already being tracked.
+		// Updates can be quite disruptive. We have to be very careful with updating
+		// resources that potentially imply disrupting customer workloads. We have
+		// to check the state of all deployments before we can safely go ahead with
+		// the update procedure.
+		for _, d := range currentDeployments {
+			allReplicasUp := allNumbersEqual(d.Status.AvailableReplicas, d.Status.ReadyReplicas, d.Status.Replicas, d.Status.UpdatedReplicas)
+			if !allReplicasUp {
+				r.logger.LogCtx(ctx, "info", fmt.Sprintf("cannot update any deployment: deployment '%s' must have all replicas up", d.GetName()))
+				return nil, nil
+			}
+		}
```

Comment:
We try to ensure the deployments are only updated in case they are healthy as far as we can tell using the Kubernetes specific API data.